src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/QueryExecutor.java

```diff
@@ -187,32 +185,29 @@ private Uni<QueryOuterClass.ResultSet> queryBridge(QueryOuterClass.Query query)
    * @param collectionName
    * @return
    */
-  protected Uni<Optional<Schema.CqlTable>> getSchema(String namespace, String collectionName) {
-    Schema.DescribeKeyspaceQuery describeKeyspaceQuery =
-        Schema.DescribeKeyspaceQuery.newBuilder().setKeyspaceName(namespace).build();
-    final Uni<Schema.CqlKeyspaceDescribe> cqlKeyspaceDescribeUni =
-        stargateRequestInfo.getStargateBridge().describeKeyspace(describeKeyspaceQuery);
-    return cqlKeyspaceDescribeUni
-        .onItemOrFailure()
-        .transformToUni(
-            (cqlKeyspaceDescribe, error) -> {
-              if (error != null
-                  && (error instanceof StatusRuntimeException sre
-                      && sre.getStatus().getCode() == Status.Code.NOT_FOUND)) {
-                return Uni.createFrom()
-                    .failure(
-                        new RuntimeException(
-                            new JsonApiException(
-                                ErrorCode.NAMESPACE_DOES_NOT_EXIST,
-                                "The provided namespace does not exist: " + namespace)));
-              }
-              Schema.CqlTable cqlTable = null;
-              return Uni.createFrom()
-                  .item(
-                      cqlKeyspaceDescribe.getTablesList().stream()
-                          .filter(table -> table.getName().equals(collectionName))
-                          .findFirst());
-            });
+  protected Uni<Optional<TableMetadata>> getSchema(String namespace, String collectionName) {
+    KeyspaceMetadata keyspaceMetadata;
+    try {
+      keyspaceMetadata =
+          cqlSessionCache
+              .getSession()
+              .getMetadata()
+              .getKeyspaces()
+              .get(CqlIdentifier.fromCql("\"" + namespace + "\""));
+    } catch (Exception e) {
+      return Uni.createFrom().failure(e);
+    }
+    // if namespace does not exist, throw error
+    if (keyspaceMetadata == null) {
+      return Uni.createFrom()
+          .failure(
+              new RuntimeException(
```

Comment:
Is there a reason to add `RuntimeException` here? `JsonApiException` is a `RuntimeException` already?
It seems better to just use plain `JsonApiException`.